### DISCUSSION TITLE
[RFC] New Variant: Steps type `inline`

### DISCUSSION THREAD

**Discussion by MadCcc** [ant-design/ant-design#38210] [RFCs - archive]
![image](https://user-images.githubusercontent.com/27722486/197767506-70ca4368-b7b5-4a71-8cf2-7bc08738757d.png)
## Summary
Steps 组件新增 `inline` 变体

## Motivation
目前 Steps 在用于页面级、区块级的步骤条样式都比较丰富，能够满足绝大多数场景的需求，但是在实际项目中我们会遇到在表格、列表等局部内容场景中等展示对象所在流程、当前状态的情况。这时使用 Steps 现有样式会存在尺寸过大、视觉过重等问题。因此可以考虑补充一种适用于内容场景的小尺寸步骤条，以满足实际业务需要。

## API

**StepsProps**

| 属性        | 说明     | 类型             | 默认值 |
| --- | --- | --- | --- |
| type | 步骤条类型 | `default` \| `navigation` \| `inline` | `default` | 

当 `type="inline"` 时，以下属性将不支持修改：
- direction（固定为 `horizontal`)
- labelPlacement (固定为 `vertical`)
- percent
- progressDot (固定为 true，即只支持点状）
- responsive （固定为 false）
- size

同时为了实现 Tooltip 功能，在 rc-steps 中为 StepItem 添加 render 方法

**StepProps**

| 属性        | 说明     | 类型             | 默认值 |
| --- | --- | --- | --- |
| render | 自定义渲染步骤 | `(stepItem: ReactNode) => ReactNode` | - | 

其中 `icon` 和 `subtitle` 不生效。

## Basic Example
![image](https://user-images.githubusercontent.com/27722486/197770484-4631dd24-67b2-4699-89ae-c0de0dd2620c.png)
```tsx
import React from React;
import { Steps, StepsProps } from 'antd';

const items: StepsProps['item'] = [
  {
    title: '开发',
  },
  {
    title: '测试',
    description: '测试阶段：测试中'
  },
  {
    title: '预发',
  },
  {
    title: '发布',
  },
];

export default () => (
  <Steps type="inline" current={2} items={items} />
);
```

### 节点类型及样式
![image](https://user-images.githubusercontent.com/27722486/197770836-94fc8115-8b26-4afd-8eec-6502d65c7619.png)
- 未开始的步骤：灰色描边圆点
- 进行中的步骤：颜色填充圆点，可以自定义状态色实现当前节点状态信息的展示
- 已完成的步骤：灰色填充圆点

其中线宽 1px，圆点尺寸 6px，未开始时边框 1px

### 尺寸及间距
![image](https://user-images.githubusercontent.com/27722486/197771089-2156d5b6-9ad5-4cd7-ab47-82a7401941d7.png)

其中每小块内间距为 4px，不定宽，但是推荐文字字数不超过三个字。热区圆角 4px，背景色取 `controlItemBgHover`。

### Hover 显示节点详细信息
![image](https://user-images.githubusercontent.com/27722486/197776117-2995c3e6-3564-4a6f-8269-83db78c54a99.png)
详细信息由 `description` 提供，利用 rc-steps 新的 render 方法实现。

## Drawbacks
No obvious drawbacks yet.

**Comment by CNFeffery** [comment]
很棒的想法👍

**Comment by JarvisArt** [comment]
1.responsive 默认为 false ?
2.StepsProps['item'] 是否支持 icon 与 subTitle ？

**Reply by MadCcc** [reply]
1. 是的
2. 不支持，我在正文里补一下

**Comment by wsxe9988** [comment]
很nice的主意
